### PR TITLE
number of saved applications now displayed on dashboard

### PR DESCRIPTION
**Pull request by Anbranin**
I did ask the Training Supervisor about this, and he says it's great.

screen shot 2017-03-21 at 3 16 26 pm

screen shot 2017-03-21 at 4 04 26 pm

### CODE REVIEW

File: app/views/dashboard/_pending_applications.haml [hunk 1]
```diff
@@ -10,5 +10,6 @@
               %li= link_to record.user.proper_name, record
           - else
             No pending applications
-        = link_to 'View saved applications for ' + position.name,
+        - count = position.saved_applications.count
```
Comment (dfaulken) [inline app/views/dashboard/_pending_applications.haml]: Should be an instance variable (the applications, at least, if not the count).

File: app/models/position.rb [hunk 2]
```diff
@@ -18,6 +18,10 @@ def name_and_department
     "#{name} (#{department.name})"
   end
 
+  def saved_applications
```
Comment (dfaulken) [inline app/models/position.rb]:
```ruby
def saved_applications
  application_records.where(saved_for_later: true)
end
```

should be okay, I think.

File: app/views/dashboard/_pending_applications.haml [hunk 3]
```diff
@@ -10,5 +10,6 @@
...
+        = link_to "View saved applications for #{position.name} (#{count})",
```
Comment (dfaulken) [inline app/views/dashboard/_pending_applications.haml]: I maintain that this should be wrapped in a conditional checking for none. If we're worried about UI elements being inconsistent, perhaps some text (`No saved applications`) in the event where there are 0?

**Comment by dfaulken** [comment]
As far as the UI goes, should the link be indented (padding) the same amount as the position name? Looks like it's a little less.

File: app/views/dashboard/_pending_applications.haml (hunk 1 above)
Comment (Anbranin) [inline app/views/dashboard/_pending_applications.haml]: Here's the thing: The `position` referenced here is something being iterated through in the view. There is currently no way to turn `position.saved_applications` into an instance variable, given that we get it from an instance variable called `@departments`.

File: app/models/position.rb (hunk 2 above)
Comment (Anbranin) [inline app/models/position.rb]: woops I forgot how 2 rails

File: app/views/dashboard/_pending_applications.haml (hunk 3 above)
Comment (Anbranin) [inline app/views/dashboard/_pending_applications.haml]: I agree with having the text "no saved applications." I did not agree with just showing the default text that it has shown for months when there are none. Will make that change.

**Comment by Anbranin** [comment]
That was a deliberate stylistic choice. I liked it.

File: app/views/dashboard/_pending_applications.haml (hunk 1 above)
Comment (dfaulken) [inline app/views/dashboard/_pending_applications.haml]: We discussed in person just now iterating through in a similar manner to pending applications.

File: app/views/dashboard/_pending_applications.haml [hunk 4]
```diff
@@ -15,4 +15,4 @@
           = link_to "View saved applications for #{position.name} (#{count})",
             saved_applications_position_path(position)
         -else
-          No saved applications
+          %ul No saved applications
```
Comment (dfaulken) [inline app/views/dashboard/_pending_applications.haml]: Does it also look consistent if there *are* saved applications?

Comment (Anbranin) [inline app/views/dashboard/_pending_applications.haml]: YES NOW IT DOES

**Comment by dfaulken** [comment]
Is this ready for review again?

**Comment by Anbranin** [comment]
YES

File: app/controllers/positions_controller.rb [hunk 5]
```diff
@@ -20,7 +20,8 @@ def destroy
   end
 
   def saved_applications
-    @saved = ApplicationRecord.where(saved_for_later: true, position: @position)
+    @saved = ApplicationRecord.where(saved_for_later: true,
```
Comment (dfaulken) [inline app/controllers/positions_controller.rb]: Or
```ruby
@position.application_records.where(saved_for_later: true)
```

File: app/controllers/sessions_controller.rb [hunk 6]
```diff
@@ -39,6 +39,6 @@ def find_user
   end
 
   def new_spire
-    (User.pluck(:spire).map(&:to_i).last + 1).to_s.rjust 8, '0'
+    ((User.pluck(:spire).map(&:to_i).last + 1).to_s.rjust 8, '0') + '@umass.edu'
```
Comment (dfaulken) [inline app/controllers/sessions_controller.rb]: The parentheses that were originally omitted were around the `rjust`, not the entire expression.

File: app/views/dashboard/_pending_applications.haml [hunk 7]
```diff
@@ -10,5 +10,9 @@
               %li= link_to record.user.proper_name, record
           - else
             No pending applications
-        = link_to 'View saved applications for ' + position.name,
-          saved_applications_position_path(position)
+        - if @saved_records[position].present?
+          - count = @saved_records[position].count
+          %ul= link_to "View saved applications for #{position.name} (#{count})",
```
Comment (dfaulken) [inline app/views/dashboard/_pending_applications.haml]: Since both cases include a `ul`, it'd be better to write it outside of the conditional.

**Comment by Anbranin** [comment]
@dfaulken ready again for review.

**Pull request merged**